.github/workflows/performance-regression.yml

```diff
@@ -12,7 +12,7 @@ jobs:
     name: Detect Regression
     needs: PR-Content-Check
     if: ${{ needs.PR-Content-Check.outputs.result == 'pass' }}
-    runs-on: ubuntu-latest
+    runs-on: macos-latest # ubuntu-latest
```

Comment:
In general, we should only use MacOS runners if we need to test specifically that something works correctly on MacOS or we're distributing a MacOS-specific binary.

What's the rationale for running on MacOS here?

Reply:
I'm trying to run more warm-ups and iterations for the benchmark-cli command in order to generate a consistent threshold value to help us identify the performance regression. However, command with 150+ executions will hit the resource limitation of the GHA linux runner.

Reply:
Which resource limit? Is it causing the job to timeout?

Reply:
https://github.com/actions/runner-images/issues/6680

Reply:
Also I found another issue that generates the wrong threshold value so it's not stable.